pkg/proxy/parser.go

```diff
@@ -423,15 +431,29 @@ func (p *Parser) parseZmodemState(b []byte) {
 
 // parseVimState 解析vim的状态，处于vim状态中，里面输入的命令不再记录
 func (p *Parser) parseVimState(b []byte) {
-	if !p.inVimState && IsEditEnterMode(b) {
-		if !isNewScreen(b) {
+	if !p.isEditMode && IsEditEnterMode(b) {
+		p.isEditMode = true
+		logger.Debugf("Session %s enter edit mode", p.id)
+	}
+	if p.isEditMode {
+		//if !p.inVimState && !p.isScreenMode {
+		//	fmt.Println("-----------hexdump---------")
+		//	fmt.Println(hex.Dump(b))
+		//}
+		if !p.isScreenMode && isNewScreen(b) {
+			p.isScreenMode = true
+			logger.Debugf("Session %s In screen state: true", p.id)
+		}
+		if !p.isScreenMode && !p.inVimState && matchMark(b, vimMarks) {
 			p.inVimState = true
-			logger.Debug("In vim state: true")
+			logger.Debugf("Session %s In vim state: true", p.id)
 		}
 	}
-	if p.inVimState && IsEditExitMode(b) {
+	if p.isEditMode && IsEditExitMode(b) {
+		p.isEditMode = false
 		p.inVimState = false
-		logger.Debug("In vim state: false")
+		p.isScreenMode = false
+		logger.Debugf("Session %s exit ( edit | vim | screen) mode", p.id)
 	}
 }
 
```

Comment:
There are no clear patterns or obvious deviations detected between the two versions of your code. They both seem to be compatible with the specifications from `2021-09-01` up until the current date (`2025-08-07`) for a parser implementation related to Telnet sessions in Go.

However, I would recommend checking specific functionalities that may have been modified or added since then against those in the previous version:

- Check for any changes regarding Vim commands handling, such as `esc]`
- Verify if there were additional checks or updates in parsing multi-command functionality
- Review the usage and modifications made specifically targeting newer features or technologies

Keep these considerations in mind while making further adjustments or improvements on the project.